src/stacked_borrows/mod.rs

```diff
@@ -377,7 +401,7 @@ impl<'tcx> Stack {
                 0
             };
             self.pop_items_after(first_incompatible_idx, |item| {
-                Stack::item_popped(&item, global, dcx)?;
+                Stack::item_popped(&item, global, dcx, /* deallocation */ false)?;
```

Comment:
I think when you feel the need to put a comment next to a bool like this, a bool is no longer the best choice. Could this be two functions now? Or an enum, perhaps:
```rust
enum PopCause {
    Deallocation,
    Aliasing,
}
```
(I am not excited by these names)

Reply:
Yeah, we have some of these and I'm just usually too lazy... I wish we had named function parameters.^^